src/test/rpc/AccountObjects_test.cpp

```diff
@@ -316,10 +316,200 @@ class AccountObjects_test : public beast::unit_test::suite
+            jvPayChan[jss::Flags] = tfUniversal;
+            jvPayChan[jss::Account] = gw.human ();
+            jvPayChan[jss::Destination] = alice.human ();
+            jvPayChan[jss::Amount] = XRP (300).value().getJson (0);
+            jvPayChan[sfSettleDelay.jsonName] = 24 * 60 * 60;
+            jvPayChan[sfPublicKey.jsonName] = strHex (gw.pk().slice ());
+            env (jvPayChan);
+            env.close();
+        }
+        {
+            // Find the payment channel.
+            Json::Value const resp = acct_objs (gw, jss::payment_channel);
+            BEAST_EXPECT (acct_objs_is_size (resp, 1));
+
+            auto const& payChan = resp[jss::result][jss::account_objects][0u];
+            BEAST_EXPECT (payChan[sfAccount.jsonName] == gw.human());
+            BEAST_EXPECT (payChan[sfAmount.jsonName].asUInt() == 300'000'000);
+            BEAST_EXPECT (payChan[sfSettleDelay.jsonName].asUInt() == 24 * 60 * 60);
+        }
+        // Make gw multisigning by adding a signerList.
+        env (signers (gw, 6, { { alice, 7} }));
+        env.close();
+        {
+            // Find the signer list.
+            Json::Value const resp = acct_objs (gw, jss::signer_list);
+            BEAST_EXPECT (acct_objs_is_size (resp, 1));
+
+            auto const& signerList = resp[jss::result][jss::account_objects][0u];
+            BEAST_EXPECT (signerList[sfSignerQuorum.jsonName] == 6);
+            auto const& entry =
+                signerList[sfSignerEntries.jsonName][0u][sfSignerEntry.jsonName];
+            BEAST_EXPECT (entry[sfAccount.jsonName] == alice.human());
+            BEAST_EXPECT (entry[sfSignerWeight.jsonName].asUInt() == 7);
+        }
+        // Create a Ticket for gw.
+        env (ticket::create (gw, gw));
+        env.close();
+        {
+            // Find the ticket.
+            Json::Value const resp = acct_objs (gw, jss::ticket);
+            BEAST_EXPECT (acct_objs_is_size (resp, 1));
+
+            auto const& ticket = resp[jss::result][jss::account_objects][0u];
+            BEAST_EXPECT (ticket[sfAccount.jsonName] == gw.human());
+            BEAST_EXPECT (ticket[sfLedgerEntryType.jsonName] == "Ticket");
+            BEAST_EXPECT (ticket[sfSequence.jsonName].asUInt() == 8);
+        }
+        // Run up the number of directory entries so gw has two
+        // directory nodes.
+        for (int d = 1'000'032; d >= 1'000'000; --d)
+        {
+            env (offer (gw, USD (1), drops (d)));
+            env.close();
+        }
+
+        // Verify that the non-returning types still don't return anything.
+        BEAST_EXPECT (acct_objs_is_size (acct_objs (gw, jss::account), 0));
+        BEAST_EXPECT (acct_objs_is_size (acct_objs (gw, jss::amendments), 0));
+        BEAST_EXPECT (acct_objs_is_size (acct_objs (gw, jss::directory), 0));
+        BEAST_EXPECT (acct_objs_is_size (acct_objs (gw, jss::fee), 0));
+        BEAST_EXPECT (acct_objs_is_size (acct_objs (gw, jss::hashes), 0));
```

Comment:
These tests are great. I'm was going to suggest you do one more unfiltered, but it looks like `testUnsteppedThenStepped` has that covered.